packages/flutter_tools/lib/src/localizations/gen_l10n.dart

```diff
@@ -146,13 +146,26 @@ String generateNumberFormattingLogic(Message message) {
       }
       final Iterable<String> parameters =
         placeholder.optionalParameters.map<String>((OptionalParameter parameter) {
-          return '${parameter.name}: ${parameter.value}';
+          if (parameter.value is num) {
+            return '${parameter.name}: ${parameter.value}';
+          } else {
+            return '${parameter.name}: ${_generateStringParameterValue(
```

Comment:
Could you explain why `_generateStringParameterValue` is necessary here? It could be helpful to add a comment above the method explaining this. 

It also seems to address some case unrelated to the number formatting? If this is indeed unrelated (but important), make sure to add a test for this to make sure that this doesn't regress

Reply:
Some formats have String parameters i.e. NumberFormat.currency has a parameter `name`. If some user passes a value with single-quote (') or double-quotes (") in the value. such as.

{
  "name": "asd'"
}

then this function will generate an optimized value i.e.
1)- `asd'` will be converted to "asd'"
2)- `asd"` will be converted to 'asd"'
3)- `asd"'` this one contains both single quote and double quotes so it will be converted as 'asd"\''

Reply:
A similar function already exists: https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/localizations/localizations_utils.dart#L267

Could you use this one instead and add a few tests covering these examples? Thanks!

Reply:
Ok, I didn't know about it, will use it now.

Reply:
Don't forget to add tests for make sure that the special cases in the string are accounted for.